kasa/smartdevice.py

```diff
@@ -554,6 +554,23 @@ def is_on(self) -> bool:
         """
         raise NotImplementedError("Device subclass needs to implement this.")
 
+    @property  # type: ignore
+    @requires_update
+    def on_since(self) -> Optional[datetime]:
+        """Return pretty-printed on-time, if available.
+
+        Returns None if the device is turned off or does not report it.
+        """
+        if "on_time" not in self.sys_info:
+            return None
+
+        if self.is_off:
+            return None
```

Comment:
Point of discussion only:

It'd be a breaking change, but what if OFF returned `False`. This way, using, abstraction you can differentiate between "device is off", and "device does not support on time"

Reply:
I was thinking about raising an exception, as I don't want to overload the meaning of this. I'm not sure how commonly this is used anyway, so I decided that None is fine. If this is really wanted, we could introduce later on `has_usage_meter` or something like this, I suppose the devices having this information will also support the usage meter (https://github.com/GadgetReactor/pyHS100/issues/113).